README.md

```diff
@@ -234,6 +234,35 @@ let greeter = { name, age, ... }: "${name} is ${toString age} years old";
 in greeter person # ... but the call works due to the ellipsis.
 ```
 
+Nix also supports binding the whole set of passed in attributes to a
+parameter using the `@` syntax:
+
+```nix
+let func = { name, age, ... }@args: builtins.attrNames args;
+in func {
+    name = "Slartibartfast";
+    age = 42;
+    email = "slartibartfast@magrath.ea";
+}
+
+# yields: [ "age" "email" "name" ]
+```
+
+**Warning:** Combining the `@` syntax with default arguments can lead
+to surprising behaviour, as the passed attributes are bound verbatim.
+This means that defaulted arguments are not included in the bound
+attribute set:
+
+```nix
+let greeter = { name, age ? 42 }@args:
+    "${args.name} is ${toString args.age} years old";
+in greeter {
+  name = "Slartibartfast";
+}
+
+# throws: error: attribute 'age' missing
```

Comment:
Think it needs a second less complex example:

```
> ({ a ? 1, b }@args: args) { b = 1; }
{ b = 1; }

> ({ a ? 1, b }@args: args) { b = 1; a = 2; }
{ a = 2; b = 1; }
```

which shows the dumb behaviour a bit better